backend/migration_tools/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,367 @@
+        else:
+            query = "SELECT id, schema_name FROM account_organization;"
+            cur.execute(query)
+
+        schema_names = [(row[0], row[1]) for row in cur.fetchall()]
+        cur.close()
+        conn.close()
+        return schema_names
+
+    def migrate_single_row(
+        self,
+        row: tuple[Any, ...],
+        dest_cursor: cursor,
+        column_names: list[str],
+        new_key_transaction: dict[str, dict[str, Any]],
+    ) -> tuple[Any, ...]:
+        row = list(row)
+
+        for key, transaction in new_key_transaction.items():
+            if key in column_names:
+                query = transaction["query"]
+                params = transaction["params"]
+                param_values = [row[column_names.index(param)] for param in params]
+
+                dest_cursor.execute(query, param_values)
+                new_key_value = dest_cursor.fetchone()
+                if new_key_value:
+                    row[column_names.index(key)] = new_key_value[0]
+
+        for i, value in enumerate(row):
+            if isinstance(value, dict):
+                row[i] = json.dumps(value)
+        return tuple(row)
+
+    def migrate_rows(
+        self,
+        migration_name: str,
+        src_cursor: cursor,
+        dest_cursor: cursor,
+        dest_conn: connection,
+        dest_query: str,
+        column_names: list[str],
+        new_key_transaction: dict[str, dict[str, Any]],
+    ) -> None:
+        batch_count = 0
+        total_rows = 0
+
+        while True:
+            rows = src_cursor.fetchmany(self.batch_size)
+            if not rows:
+                break
+
+            batch_count += 1
+            total_rows += len(rows)
+            logger.info(
+                f"[{migration_name}] Processing batch {batch_count}: {len(rows)} rows"
+            )
+
+            converted_rows = []
+            for row in rows:
+                converted_row = self.migrate_single_row(
+                    row, dest_cursor, column_names, new_key_transaction
+                )
+                converted_rows.append(converted_row)
+
+            start_time = time.time()
+            try:
+                dest_cursor.executemany(dest_query, converted_rows)
+                dest_conn.commit()
+                end_time = time.time()
+                elapsed_time = end_time - start_time
+                logger.info(
+                    f"[{migration_name}] Batch {batch_count} inserted successfully."
+                    f" Time taken: {elapsed_time:.2f} seconds."
+                )
+            except psycopg2.Error as e:
+                dest_conn.rollback()
+                logger.error(
+                    f"[{migration_name}] Error inserting batch {batch_count}: {e}"
+                )
+                raise
+        logger.info(
+            f"[{migration_name}] Completed migration with {total_rows} rows processed."
+        )
+
+    def adjust_auto_increment(
```

Comment:
NIT: Rename this to be more explicit or consider adding a docstring

Reply:
Done.